press-release-faq.md

```diff
@@ -0,0 +1,176 @@
+# Vision PR/FAQ
+
+> ⚠️ This is a mock press release / FAQ for the eventual launch of BWE, it does not necessarily represent
+the current state of the project and the contents here may change during the course of implementation.
+It is used for communicating the vision of the project and maintaining alignment on the criteria for success.
+
+## Press Release
+
+Pagoda is proud to announce the production release of the BOS Web Engine (BWE).
+
+BWE is the new execution environment for BOS components which offers a more reliable security guarantee and
+enables modern web development features which have previously been missing like npm dependencies, TypeScript,
+and hooks— along with advanced features for performance optimizations.
+
+Until now, BOS components have been written to be compatible with the "VM". This VM processes components line
+by line and executes statements only if it has been built to understand them and knows them to be safe. It
+relies on this restrictive processing for security. This model creates a constant tug of war between expanding the
+capability of BOS components and maintaining security. Importing code which expects to execute in a typical
+browser environment is not possible.
+
+In contrast, BWE leverages the sandboxing configuration of iframes built into browsers to isolate the execution
+of each component. Each component's code is passed to an invisible iframe along with an instance of Preact. Preact
+executes the components and the rendered output is serialized and passed out of the iframe. Then, the output of all
+the iframes is composed together into a single tree for render on the page. One way to think of this is that each
+component is run as its own isolated applet and given a messaging system to integrate with other components. This
+allows being significantly less restrictive with the code a component can run, since the only surface of exposure
+to the OWA is DOM nodes. It also means code written for a standard browser or Preact environment can be imported
+and run by components.
```

Comment:
```suggestion
BWE takes an alternative approach by leveraging the inherent sandboxing feature of iframes in web browsers to isolate the execution of each component.

Here's how it works:

- Code for each component is sent to a hidden iframe, accompanied by an instance of [Preact](https://preactjs.com/), a lightweight JavaScript library for building user interfaces.
- Within this iframe, Preact executes the component's code, and once rendered, the output is systematically converted into a secure format that is transferred out of the iframe.
- The outputs from all iframes are then meticulously assembled into one unified structure, ready to be displayed on the webpage.

Imagine each component operating as a self-contained mini-application, complete with its own communication system to synchronize with other components. This architecture allows for significantly fewer restrictions with code components that can execute since the only interaction point with the main web application is the DOM nodes. Additionally, code designed for standard browsers or Preact can be seamlessly imported and executed within these components.
```